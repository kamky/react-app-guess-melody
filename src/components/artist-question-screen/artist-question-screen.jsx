// React
import React from "react";

// Prop Types
import PropTypes from "prop-types";
import {ArtistQuestionTypes} from "./artist-question-screen.prop";

// Components
import {ArtistItem} from "../artist-item/artist-item";

// Component ArtistQuestionScreen
export const ArtistQuestionScreen = (props) => {
  const {question, onUserAnswer, renderAudioPlayer} = props;
  const {answers, song} = question;

  const handleUserAnswer = (answer) => {
    onUserAnswer(answer, question);
  };

  return (
    <section className="game__screen">
      <h2 className="game__title">Кто исполняет эту песню?</h2>
      <div className="game__track">
        <div className="track">
          {renderAudioPlayer(song.src, 0)}
        </div>
      </div>

      <form className="game__artist">
        {answers.map((answer, idx) => (
          <ArtistItem
            key={`${idx}-${answer.artist}`}
            index={idx}
            answer={answer}
            onUserAnswer={handleUserAnswer}
          />
        ))}
      </form>
    </section>
  );
};

ArtistQuestionScreen.propTypes = {
  question: ArtistQuestionTypes,
  onUserAnswer: PropTypes.func.isRequired,
  renderAudioPlayer: PropTypes.func.isRequired,
};
