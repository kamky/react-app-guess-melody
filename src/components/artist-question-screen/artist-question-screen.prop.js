// Types
import PropTypes from "prop-types";
import {gameTypes} from "../../consts";

export const ArtistQuestionTypes = PropTypes.shape({
  type: PropTypes.oneOf([gameTypes.ARTIST]).isRequired,
  song: PropTypes.shape({
    artist: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
  }).isRequired,
  answers: PropTypes.arrayOf(
      PropTypes.shape({
        picture: PropTypes.string.isRequired,
        artist: PropTypes.string.isRequired,
      }).isRequired
  ).isRequired,
}).isRequired;
