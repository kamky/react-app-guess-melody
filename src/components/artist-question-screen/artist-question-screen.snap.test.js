// React
import React from "react";
import renderer from "react-test-renderer";

// Components
import {ArtistQuestionScreen} from "./artist-question-screen";

const questionMock = {
  type: `artist`,
  song: {
    artist: `artist-1`,
    src: `src-1`,
  },
  answers: [
    {
      picture: `picture-1`,
      artist: `artist-1`,
    },
    {
      picture: `picture-2`,
      artist: `artist-2`,
    },
    {
      picture: `picture-3`,
      artist: `artist-3`,
    },
  ],
};

describe(`Snapshot test for component <ArtistQuestionScreen />`, () => {
  const question = questionMock;
  const defaultProps = {
    question,
    onUserAnswer: jest.fn(),
    renderAudioPlayer: jest.fn(),
  };

  const wrapper = renderer.create(<ArtistQuestionScreen {...defaultProps} />);
  test(`should component <ArtistQuestionScreen /> renders correctly`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
