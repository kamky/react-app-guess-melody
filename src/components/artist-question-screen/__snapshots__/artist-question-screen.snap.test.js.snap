// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for component <ArtistQuestionScreen /> should component <ArtistQuestionScreen /> renders correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
    </div>
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="artist-1"
          className="artist__picture"
          src="picture-1"
        />
        artist-1
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="artist-2"
          className="artist__picture"
          src="picture-2"
        />
        artist-2
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="artist-3"
          className="artist__picture"
          src="picture-3"
        />
        artist-3
      </label>
    </div>
  </form>
</section>
`;
